pipdecode.js

```diff
@@ -0,0 +1,32 @@
+require('buffertools').extend();
```

Comment:
have you seen https://github.com/rgbkrk/pipboylib/blob/master/lib/parser.js

You seem to have implemented way more protocol though!

The api I would eventually like to have would allow piping the relay into the parser into websocket/filters/whatever into serialiser into relay.